Weather Man/weather_calculator.py

```diff
@@ -0,0 +1,116 @@
+"""
+weather_calculator.py
+
+This module defines the WeatherCalculator class, which performs calculations on weather data to generate yearly extremes,
+monthly averages, and daily temperatures.
+
+Classes:
+- WeatherCalculator: A class to calculate weather extremes, averages, and daily temperatures based on provided weather data.
+
+"""
+
+from datetime import datetime
+from constants import weather_calculation_results
+
+
+class WeatherCalculator:
+    """
+    A class to perform calculations on weather data and generate reports.
+
+    Attributes:
+    - weather_data: Weather data containing temperature and humidity information.
+
+    Methods:
+    - calculate_yearly_extremes(year): Calculates yearly extremes (highest temperature, lowest temperature,
+                                        most humid day) for the specified year.
+    - calculate_monthly_averages(year, month): Calculates monthly averages (highest temperature, lowest temperature,
+                                                mean humidity) for the specified year and month.
+    - calculate_daily_temperatures(year, month): Retrieves daily temperatures for the specified year and month.
+    """
+
+    def __init__(self, weather_data):
+        """
+        Initializes WeatherCalculator with weather data.
+
+        Args:
+        - weather_data (list): List of weather data objects containing temperature and humidity information.
+        """
+        self.weather_data = weather_data
+
+    def calculate_yearly_extremes(self, year):
+        """
+        Calculates yearly extremes (highest temperature, lowest temperature, most humid day) for the specified year.
+
+        Args:
+        - year (int): Year for which extremes are calculated.
+
+        Returns:
+        - WeatherCalculationResults: Object containing calculated extremes.
+        """
+        year_data = self.weather_data
+
+        if not year_data:
+            print("No data found for the year")
+            return weather_calculation_results
+
+        highest_temp = max(year_data, key=lambda x: x.max_temperature)
+        lowest_temp = min(year_data, key=lambda x: x.min_temperature)
+        most_humid_day = max(year_data, key=lambda x: x.max_humidity)
```

Comment:
Instead of running 3 different lambda function, Can we use a single loop and calculate the required values